src/App.js

```diff
@@ -1,12 +1,101 @@
 import React from 'react';
 import './App.css';
+import PostList from './components/PostList/PostList';
+import { getComments } from './api/getComments';
+import { getPosts } from './api/getPosts';
+import { getUsers } from './api/getUsers';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Dynamic list of posts</h1>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    posts: [],
+    copyPosts: [],
+    users: [],
+    comments: [],
+    isLoading: false,
+  };
+
+  dataLoading = () => {
+    this.setState({
+      isLoading: true,
+    });
+
+    Promise.all([getComments(), getUsers(), getPosts()]).then(data => this.setState({
+      comments: data[0],
+      users: data[1],
+      posts: data[2],
+      copyPosts: data[2],
+      isLoading: false,
+    }));
+  };
+
+  searchPost = ({ target }) => {
+    this.setState(prevState => ({
+      posts: prevState.copyPosts.filter(
+        post => post.title.toLowerCase().includes(target.value.toLowerCase())
+          || post.body.toLowerCase().includes(target.value.toLowerCase())
+      ),
+    }));
+  };
+
+  render() {
+    const {
+      posts, users, comments, copyPosts,
+    } = this.state;
+    const preparedPosts = UserToPosts(posts, users);
```

Comment:
функция должна нести смысловое значение в названии
```suggestion
    const preparedPosts = getPostsWithUser(posts, users);
```